tests/experimental/pydantic/test_conversion.py

```diff
@@ -645,13 +661,153 @@ class UserInput:
         age: strawberry.auto
         password: strawberry.auto
 
-    data = UserInput(1)
+    data = UserInput(age=1)
     user = data.to_pydantic()
 
     assert user.age == 1
     assert user.password is None
 
 
+def test_can_convert_input_types_to_pydantic_default_values_defaults_declared_first():
+    # test that we can declare a field with a default. before a field without a default
+    class User(pydantic.BaseModel):
+        password: Optional[str] = None
+        age: int
+
+    @strawberry.experimental.pydantic.input(User)
+    class UserInput:
+        password: strawberry.auto
+        age: strawberry.auto
+
+    data = UserInput(age=1)
+    user = data.to_pydantic()
+
+    assert user.age == 1
+    assert user.password is None
+
+    definition: TypeDefinition = UserInput._type_definition
+    assert definition.name == "UserInput"
+
+    [
+        age_field,
+        password_field,
+    ] = (
+        definition.fields
+    )  # fields without a default go first, so the order gets reverse
+
+    assert age_field.python_name == "age"
+    assert age_field.type is int
+
+    assert password_field.python_name == "password"
+    assert isinstance(password_field.type, StrawberryOptional)
+    assert password_field.type.of_type is str
+
+
+def test_sort_creation_fields():
+    has_default = DataclassCreationFields(
+        name="has_default",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default",
+            graphql_name="has_default",
+            default="default_str",
+            default_factory=UNSET,
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    has_default_factory = DataclassCreationFields(
+        name="has_default_factory",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default_factory",
+            graphql_name="has_default_factory",
+            default=UNSET,
+            default_factory=lambda: "default_factory_str",
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    no_defaults = DataclassCreationFields(
+        name="no_defaults",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="no_defaults",
+            graphql_name="no_defaults",
+            default=UNSET,
+            default_factory=UNSET,
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    fields = [has_default, has_default_factory, no_defaults]
+    assert sort_creation_fields(fields) == [
+        no_defaults,
+        has_default,
+        has_default_factory,
+    ]  # should place items with defaults last
+
+
+def test_defaults_into_factory():
+    assert (
+        defaults_into_factory(default=arguments.UNSET, default_factory=arguments.UNSET)
+        is arguments.UNSET
+    )  # should return UNSET when both defaults are UNSET
```

Comment:
Couple minor comments here:
1. You added `from arguments import UNSET` at the top of the file, so no reason to grab it from the `arguments` namespace anymore
2. Just add the comments above the line they refer to. The multi-line assert with trailing in-line comments is pretty strange

Reply:
good catch! resolved :)